src/detect-engine-dns.c

```diff
@@ -119,6 +119,65 @@ int DetectEngineInspectDnsQueryName(ThreadVars *tv,
     return r;
 }
 
+int DetectEngineInspectDnsResponse(ThreadVars *tv,
+                                    DetectEngineCtx *de_ctx, DetectEngineThreadCtx *det_ctx,
+                                    const Signature *s, const SigMatchData *smd,
+                                    Flow *f, uint8_t flags, void *alstate,
+                                    void *txv, uint64_t tx_id)
+{
+    uint8_t *buffer;
+    uint32_t buffer_len;
+    int r = 0;
+
+    SCLogDebug("start");
+
+#ifdef HAVE_RUST
+    for (uint16_t i = 0;; i++){
+        det_ctx->discontinue_matching = 0;
+        det_ctx->buffer_offset = 0;
+        det_ctx->inspection_recursion_counter = 0;
+
+        if (rs_dns_tx_get_answer_rdata(txv, i, &buffer, &buffer_len)) {
+            r = DetectEngineContentInspection(de_ctx, det_ctx,
+                s, smd, f, buffer, buffer_len, 0,
+                DETECT_ENGINE_CONTENT_INSPECTION_MODE_STATE, NULL);
+            if (r == 1) {
+                break;
+            }
+        } else {
+            break;
+        }
+    }
+#else
+
+    DNSTransaction *tx = (DNSTransaction *)txv;
+    DNSAnswerEntry *answer = NULL;
+
+    TAILQ_FOREACH(answer, &tx->answer_list, next) {
+        SCLogDebug("tx %p response %p", tx, answer);
+
+        det_ctx->discontinue_matching = 0;
+        det_ctx->buffer_offset = 0;
+        det_ctx->inspection_recursion_counter = 0;
+
+        buffer = (uint8_t *) ((uint8_t *) answer +
+                    sizeof(DNSAnswerEntry) + answer->fqdn_len);
+        buffer_len = answer->fqdn_len;
```

Comment:
I think the actual data len is 'answer->data_len'. See the logging code, for example https://github.com/OISF/suricata/blob/master/src/output-json-dns.c#L509

Reply:
Yeah, that makes sense. Ill submit a new PR.